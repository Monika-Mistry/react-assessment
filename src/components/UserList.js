import React, {Component} from 'react';

import User from './User';

export default class UserList extends Component {
    componentDidMount(){

    }
    render(){
        return(
            <ul>
                <User />
            </ul>
        )
    }
}