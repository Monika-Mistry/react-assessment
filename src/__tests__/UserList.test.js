import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

import UserList from '../components/UserList';

const users = [{
_id: 1,
username: "a"
}]

it('render a userlist', () => {
    const wrapper = renderer.create(<UserList users={users} getAllUser={() => console.log("hello")}/>).toJSON();
    expect(wrapper).toMatchSnapshot();
})


it('render a table', () => {
    const wrapper = mount(<UserList users={users} getAllUser={() => console.log("hello")}/>);
    const content = <th>Username</th>;
    expect(wrapper.contains(content)).toEqual(true);
})
