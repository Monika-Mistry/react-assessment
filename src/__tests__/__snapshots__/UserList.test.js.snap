// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a userlist 1`] = `
<div
  className="container"
>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Username
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          a
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
